File: problems/parker/Prob03.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package prob03;
import java.io.*;
import java.util.*;
import java.lang.*;
import java.math.*;


public class Prob03{

    public static void main(String... args) throws IOException {
     String problem = "Prob03.in.txt";
     String path = "C:\\Users\\pwatts\\Documents\\NetBeansProjects\\practice\\src\\JudgingInputs\\" + problem;
     File file = new File(path);
     
     boolean leap;
     Scanner scanner = new Scanner(file);
     scanner.nextLine();
     int caseAmount;
     int year;
     
     while(scanner.hasNextInt()){
         caseAmount = scanner.nextInt();
         for(int i = 0; i < caseAmount; i++){
             year = scanner.nextInt();
       
             if(year < 1582){
                 leap = false;
             System.out.println("No");}
            else if(!(year%4 == 0)){
                 leap = false;
                 System.out.println("No");
             }
            else if(!(year%100 == 0)){
                    leap = true; 
                    System.out.println("Yes");
                     }
            else if(!(year%400 == 0)){
                 leap = false;
                 System.out.println("No");
          
                         
                         }
             else{
                 leap = true;
                 System.out.println("Yes");
             
             }
         
            
            }
        
         }
     }
     
     
     
        
        
    }
    

File: problems/rice_py/prob03.py
import path

path = path.ex_input + "Prob03.in.txt"
file = open(path, "r")
amt = int(file.readline())

for i in range(0, amt):
	sub_amt = int(file.readline())
	for i2 in range(0, sub_amt):
		y = int(file.readline())
		if (y < 1582):
			print("No")
		elif (y % 4 != 0):
			print("No")	
		elif (y % 100 != 0):
			print("Yes")
		elif (y % 400 != 0):
			print("No")
		else:
			print("Yes")
